sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/screens/home/adapter/ItemListAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package co.nimblehq.sample.xml.ui.screens.home.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.*
+import co.nimblehq.sample.xml.databinding.ItemHomeListBinding
+import co.nimblehq.sample.xml.model.UiModel
+
+class ItemListAdapter : ListAdapter<UiModel, ItemListAdapter.ItemViewHolder>(ItemDiffCallback()) {
+
+    var onItemClicked: ((uiModel: UiModel) -> Unit)? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
+        val binding =
+            ItemHomeListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ItemViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
+        holder.bind(getItem(position), onItemClicked)
+    }
+
+    class ItemViewHolder(private val binding: ItemHomeListBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(item: UiModel, onItemClicked: ((uiModel: UiModel) -> Unit)?) {
+            binding.tvItemHome.text = item.id
+            binding.lItemHome.setOnClickListener { onItemClicked?.invoke(item) }
```

Comment:
(Optional) Use `with(binding)`

Reply:
Since there are only 2 binding here. I think we could keep it to be simple.